internal/plugins/ansible/v1/init.go

```diff
@@ -151,25 +152,41 @@ func (p *initSubcommand) PostScaffold() error {
 }
 
 // addInitCustomizations will perform the required customizations for this plugin on the common base
-func addInitCustomizations(projectName string) error {
+func addInitCustomizations(projectName string, componentConfig bool) error {
 	managerFile := filepath.Join("config", "manager", "manager.yaml")
+	managerProxyPatchFile := filepath.Join("config", "default", "manager_auth_proxy_patch.yaml")
 
 	// todo: we ought to use afero instead. Replace this methods to insert/update
 	// by https://github.com/kubernetes-sigs/kubebuilder/pull/2119
 
 	// Add leader election
-	err := util.InsertCode(managerFile,
-		"--leader-elect",
-		fmt.Sprintf("\n        - --leader-election-id=%s", projectName))
-	if err != nil {
-		return err
-	}
-	managerProxyPatchFile := filepath.Join("config", "default", "manager_auth_proxy_patch.yaml")
-	err = util.InsertCode(managerProxyPatchFile,
-		"- \"--leader-elect\"",
-		fmt.Sprintf("\n        - \"--leader-election-id=%s\"", projectName))
-	if err != nil {
-		return err
+	if componentConfig {
+		err := util.InsertCode(managerFile,
+			"- /manager",
+			fmt.Sprintf("\n        args:\n        - --leader-election-id=%s", projectName))
+		if err != nil {
+			return err
+		}
+
+		err = util.InsertCode(managerProxyPatchFile,
+			"memory: 64Mi",
+			fmt.Sprintf("\n      - name: manager\n        args:\n        - \"--leader-election-id=%s\"", projectName))
+		if err != nil {
+			return err
+		}
+	} else {
+		err := util.InsertCode(managerFile,
+			"--leader-elect",
+			fmt.Sprintf("\n        - --leader-election-id=%s", projectName))
+		if err != nil {
+			return err
+		}
+		err = util.InsertCode(managerProxyPatchFile,
+			"- \"--leader-elect\"",
+			fmt.Sprintf("\n        - \"--leader-election-id=%s\"", projectName))
+		if err != nil {
+			return err
+		}
```

Comment:
Yup.

Reply:
added step by step explanation

```
      1. If `--component-config` flag is passed then look for `- /manager` in the file in order to add leader election id into the manager.yaml file. If flag is not passed then look for `--leader-elect`.
      2. Same logic for manager proxy patch file, look for `memory: 64Mi` if flag is passed. If flag is not passed then look for `--leader-elect`
      3. Repeat the same logic for replacing the port number from the file.
      4. At the end, update `manager.yaml` file to avoid lint errors.
```